flaky/_flaky_plugin.py

```diff
@@ -75,14 +80,19 @@ def _handle_test_error_or_failure(self, test, err):
         flaky = self._get_flaky_attributes(test)
 
         if not self._has_flaky_test_failed(flaky):
-            max_runs = flaky[FlakyNames.MAX_RUNS]
-            runs_left = max_runs - flaky[FlakyNames.CURRENT_RUNS]
-            message = self._retry_failure_message.format(
-                runs_left,
-                max_runs,
-            )
-            self._log_test_failure(name, err, message)
-            self._rerun_test(test)
+            if self._should_rerun_test(test, name, err):
```

Comment:
This _log method is already uber-long and nested. Instead of it getting longer... can there be a bit of refactoring to improve readability and maintainability?